### PR TITLE
Adds user roles relation

### PR DESCRIPTION
**Pull request by scmountain**
Rspec 22-0

### CODE REVIEW

File: spec/features/user_has_roles_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require "rails_helper"
+
+describe "An user" do
+ it "has a registered user role" do
+   user = create(:user)
+   role = Role.create(name: "registered_user")
+   user.roles << role
+   visit login_path
+
```
Comment (CjMoore) [inline spec/features/user_has_roles_spec.rb]: We need to make sure the role of registered user is automatically added when a user is created as well.

File: db/schema.rb [hunk 2]
```diff
@@ -35,6 +35,28 @@
...
+    t.index ["role_id"], name: "index_user_roles_on_role_id", using: :btree
+    t.index ["user_id"], name: "index_user_roles_on_user_id", using: :btree
+  end
+
+  create_table "user_roles_tables", force: :cascade do |t|
+    t.integer "user_id"
+    t.integer "role_id"
+    t.index ["role_id"], name: "index_user_roles_tables_on_role_id", using: :btree
+    t.index ["user_id"], name: "index_user_roles_tables_on_user_id", using: :btree
+  end
```
Comment (CjMoore) [inline db/schema.rb]: looks like you guys have some duplicate

File: spec/features/guest_can_create_account_spec.rb [hunk 3]
```diff
@@ -17,5 +17,6 @@
     click_on "Create Account"
 
     expect(current_path).to eq(root_path)
+    expect(User.first.roles.first.name).to eq("registered_user")
```
Comment (CjMoore) [inline spec/features/guest_can_create_account_spec.rb]: 🎉

**Comment by edilenedacruz** [comment]
Merging. 🌟

**Pull request merged**